common/component_interface_utils/include/component_interface_utils/rclcpp/create_interface.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COMPONENT_INTERFACE_UTILS__RCLCPP__CREATE_INTERFACE_HPP_
+#define COMPONENT_INTERFACE_UTILS__RCLCPP__CREATE_INTERFACE_HPP_
+
+#include <component_interface_utils/rclcpp/service_server.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <utility>
+
+namespace component_interface_utils
+{
+
+// Use a node pointer because shared_from_this cannot be used in constructor.
+template <class SpecT, class NodeT, class CallbackT>
+typename Service<SpecT>::SharedPtr create_service_impl(
```

Comment:
I think this function is expected to be used by many developers.
So could you write some brief descriptions for each function so that developers can understand the differences between the original `rclcpp`'s functions?

Reply:
Also, writing a (short) README and explaining that these files extend `rclcpp` files would be good.

Reply:
Thank you. I added a readme and comments. https://github.com/autowarefoundation/autoware.universe/pull/704/commits/99264a40ae2be2f025df03d61cf7092702817353

Reply:
Could you link somewhere in https://github.com/ros2/rclcpp/tree/master/rclcpp?
Otherwise, we can't properly review this code. :cry:

Reply:
Like here?
https://github.com/ros2/rclcpp/blob/master/rclcpp/include/rclcpp/node_impl.hpp#L139-L154

Reply:
Yeah, exactly!

Reply:
Add comment. https://github.com/autowarefoundation/autoware.universe/pull/704/commits/f65eecfa7410f9886527d09211856661e51f0e51